client/src/storage/remote-storage.ts

```diff
@@ -58,7 +63,7 @@ export class RemoteStorage implements IStorage {
         })
     }
 
-    private async fetchResource(resourcePath: string): Promise<any | undefined> {
+    private async fetchResource<T>(resourcePath: string): Promise<T | undefined> {
```

Comment:
Ha nice! Forgot that generics were a thing in Typescript

Reply:
I've been typescripting now for a couple of years.  Must say it's not a lot less fun than Scala.